LiteLLM Minor Fixes & Improvements (10/01/2024)

krrishdholakia opened the pull request:
## Title

- fix(proxy_server.py): fix regression for prometheus callback when using 'success'/'failure' callbacks
- expose new general settings flag `disable_prisma_schema_update` - prevent proxy from updating db schema. 

## Relevant issues

<!-- e.g. "Fixes #000" -->

## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring
📖 Documentation
🚄 Infrastructure
✅ Test

## Changes

<!-- List of changes -->

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
If UI changes, send a screenshot/GIF of working UI fixes

<!-- Test procedure -->

Commit:
fix(proxy_server.py): fix regression for prometheus callback when using 'success'/'failure' callbacks

ishaan-jaff commented on litellm/proxy/example_config_yaml/otel_test_config.yaml:
```diff
@@ -36,7 +36,9 @@ model_list:
 litellm_settings:
   cache: true
-  callbacks: ["otel", "prometheus"]
+  callbacks: ["otel"]
```
this is intentional and this test was written because a specific user uses `callbacks: ["prometheus"]` 

Can you add a diff test that does `success_callback: ["prometheus"]  ` @krrishdholakia

krrishdholakia replied:
Oh! got it

krrishdholakia replied:
fixed - https://github.com/BerriAI/litellm/pull/6005/commits/36ad126c5dc8f39c61d33dd74f49cda8c01670a9

Commit:
feat(proxy/utils.py): expose a 'disable_schema_update' flag.

Optional Flag - Prevents proxy from running db migrations. Only output schema diff as sql queries.

Commit:
fix(proxy_cli.py): new flag - 'disable_prisma_schema_update' - prevents proxy from doing db migrations, just emits sql queries

Commit:
fix(proxy_server.py): add unit test for making sure prometheus is correctly added

Commit:
Merge branch 'main' into litellm_dev_10_01_2024

Commit:
fix(proxy_server.py): fix linting errors

Commit:
test: fix tests

Commit:
fix(user_api_key_auth.py): fix none handling

Commit:
test: skip expired token

Commit:
fix(user_api_key_auth.py): safely handle headers
